src/utils/memoization.js

```diff
@@ -0,0 +1,289 @@
+  return compareObjects(prevData, nextData, {
+    comparators: {
+      conferenceSolution: (...args) =>
+        compareObjects(...args, { comparators: { key: compareObjects } }),
+      createRequest: (...args) =>
+        compareObjects(...args, {
+          comparators: {
+            conferenceSolutionKey: compareObjects,
+            status: compareObjects,
+          },
+        }),
+      entryPoints: (...args) => compareLists(...args, compareObjects),
+    },
+  })
+}
+
+function compareExtendedProperties(prevProperties, nextProperties) {
+  return compareObjects(prevProperties, nextProperties, {
+    comparators: {
+      shared: compareObjects,
+      private: compareObjects,
+    },
+  })
+}
+
+function compareReminders(prevReminders, nextReminders) {
+  return compareObjects(prevReminders, nextReminders, {
+    comparators: {
+      overrides: (...args) => compareLists(...args, compareObjects),
+    },
+  })
+}
+
+function compareReschedules(prevReschedules, nextReschedules) {
+  return compareObjects(prevReschedules, nextReschedules, {
+    comparators: {
+      tracked_changes: (...args) => compareLists(...args, compareObjects),
+    },
+  })
+}
+
+function compareTags(prevTag, nextTag) {
+  return compareObjects(prevTag, nextTag, {
+    comparators: {
+      rules: (...args) =>
+        compareObjects(...args, {
+          comparators: {
+            fields: (...args) => compareLists(...args, compareObjects),
+          },
+        }),
+      user_tokens: compareUnorderedLists,
+    },
+  })
+}
+
+function compareWorkingLocationProperties(prevProperties, nextProperties) {
+  return compareObjects(prevProperties, nextProperties, {
+    comparators: {
+      customLocation: compareObjects,
+      officeLocation: compareObjects,
+    },
+    logDifferences: DEBUG_MEMOIZATION,
+    from: 'compareWorkingLocationProperties',
+  })
+}
+
+/**
+ * Check if two events are the same by matching the user_event_id or id properties.
+ * @param {Object} eventA
+ * @param {Object} eventB
+ */
+export function matchEvent(eventA, eventB) {
+  if (eventA.user_event_id) {
+    return eventA.user_event_id === eventB.user_event_id
+  }
+
+  if (eventA.id) {
+    return eventA.id === eventB.id
+  }
+
+  return false
+}
+
+/**
+ * Checks if two lists events have the same events in the array, regardless of order.
+ * When comparing events, we use our custom event comparator.
+ * @param {Object[]} prevEvents
+ * @param {Object[]} nextEvents
+ */
+function compareEventLists(prevEvents, nextEvents) {
+  return compareUnorderedLists(
+    prevEvents,
+    nextEvents,
+    matchEvent,
+    compareEvents
+  )
+}
+
+/**
+ * Compares two lists, accounting for the order of the items.
+ * @template T
+ * @param {T[]} listA
+ * @param {T[]} listB
+ * @param {(itemA: T, itemB: T) => boolean} comparator
+ */
+export function compareLists(listA, listB, comparator = Object.is) {
+  if (listA === listB) {
+    return true
+  }
+
+  if (listA.length !== listB.length) {
```

Comment:
is it dangerous to compare lists here without checking for null/undefined? could cause `TypeError`